pkg/provider/webhook/webhook_test.go

```diff
@@ -353,6 +363,64 @@ func TestWebhookGetSecret(t *testing.T) {
 	}
 }
 
+var testCasesPushSecret = `
+case: good json
+args:
+  url: /api/pushsecret?id={{ .remoteRef.remoteKey }}&secret={{ .remoteRef.secretKey }}
+  key: testkey
+  secretkey: secretkey
+  pushsecret: true
+  secret:
+    name: test-secret
+    data:
+      secretkey: value
+want:
+  path: /api/pushsecret?id=testkey&secret=secretkey
+  err: ''
+---
```

Comment:
I really don't like this yaml-based test description (always have :see_no_evil:), i wanted to move away from it, but never took the time to do so :melting_face: 

So this is fine, it is what it is. :smile:

Reply:
Yeah, me too... I almost re-wrote the whole thing. :D But it would have taken a lot more time sadly. :(